core/src/test/scala/org/apache/spark/SharedSparkContext.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2019 PingCAP, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark
+
+import org.scalatest.{BeforeAndAfterAll, BeforeAndAfterEach, Suite}
+import org.apache.spark.SharedSparkContext._
+
+trait SharedSparkContext extends BeforeAndAfterAll with BeforeAndAfterEach { self: Suite =>
+
+  def sc: SparkContext = _sc
+
+  protected var conf: SparkConf = new SparkConf(false)
+
+  protected def initializeContext(): Unit = {
+    if (null == _sc) {
+      _sc = new SparkContext("local[4]", "test", conf)
```

Comment:
why `local[4]`?

Reply:
same as Spark Core test

Reply:
cannot find `local[4]` in current code base

Reply:
https://github.com/apache/spark/blob/master/core/src/test/scala/org/apache/spark/SharedSparkContext.scala#L42